kv/kvserver: TestLeaseQueueShedsOnIOOverload failed

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/496ed085-5efa-4899-9d17-1e401fcb298c?testReportRun=1&testReportShard=21&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-24.1 @ [33ccef64d1b4cdff32c5ea3ffb57e839579e05d7](https://github.com/cockroachdb/cockroach/commits/33ccef64d1b4cdff32c5ea3ffb57e839579e05d7):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload4098329188
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:359: condition failed to evaluate within 3m45s: from lease_queue_test.go:365: expected 0 leases on store 1, found 24
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload4098329188
--- FAIL: TestLeaseQueueShedsOnIOOverload (257.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=21</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #146354 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot T-kv X-duplicate branch-release-24.3]
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-50563

--- a/pkg/kv/kvserver/lease_queue_test.go
+++ b/pkg/kv/kvserver/lease_queue_test.go
@@ -343,6 +343,11 @@ func TestLeaseQueueRaceReplicateQueue(t *testing.T) {
 func TestLeaseQueueShedsOnIOOverload(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	// The SucceedsSoon has been observed to occasionally time out in both
+	// deadlock and race builds.
+	skip.UnderDuressWithIssue(t, 138903)
+
 	ctx := context.Background()
 
 	tc := testcluster.StartTestCluster(t, 3, base.TestClusterArgs{})

